src/js/app.js

```diff
@@ -1,6 +1,270 @@
+
+
+
+// Функція для генерації випадкового значення зі списку
+function getRandomValue(list) {
+    const randomIndex = Math.floor(Math.random() * list.length);
+    return list[randomIndex];
+}
+
+
+// Функція для об'єднання об'єктів із позбавленням повторів
+function mergeObjects(arr1, arr2) {
+    const mergedObjects = [];
+
+    // Об'єдную об'єкти з першого масиву
+    for (const obj of arr1) {
+        if (obj.full_name) {
+            const duplicate = mergedObjects.find((item) => item.full_name === obj.full_name);
+            if (!duplicate) {
+                mergedObjects.push(obj);
+            }
+        }
+    }
+
+    // Об'єдную об'єкти з другого масиву
+    for (const obj of arr2) {
+        if (obj.full_name) {
+            const duplicate = mergedObjects.find((item) => item.full_name === obj.full_name);
+            if (!duplicate) {
+                mergedObjects.push(obj);
+            }
+        }
+    }
+    return mergedObjects;
+}
+//Створюємо юзера з правильно переданими параметрами
+function formatUser(user) {
+    let formatedUser = {
+        gender: user.gender,
+        title: user.name.title,
+        full_name: user.name.first + " " + user.name.last,
+        city: user.location.city,
+        state: user.location.state,
+        country: user.location.country,
+        postcode: user.location.postcode,
+        coordinates: user.location.coordinates,
+        timezone: user.location.timezone,
+        email: user.email,
+        b_date: user.dob.date,
+        age: user.dob.age,
+        phone: user.phone,
+        picture_large: user.picture.large,
+        picture_thumbnail: user.picture.thumbnail,
+        id: "",
+        favorite: false,
+        course: getRandomValue([
+            "Mathematics",
+            "Physics",
+            "English",
+            "Computer Science",
+            "Dancing",
+            "Chess",
+            "Biology",
+            "Chemistry",
+            "Law",
+            "Art",
+            "Medicine",
+            "Statistics",
+        ]),
+        bg_color: "",
+        note: "",
+
+    }
+
+    return formatedUser;
+}
+//чи валідний об'єкт user
+function isValid(user) {
+    return validString(user.full_name) &&
+        validString(user.city) &&
+        validString(user.state) &&
+        validString(user.country) &&
+        checkEmail(user.email) &&
+        checkAge(user.age) &&
+        validatePhone(user.phone, user.country)
+        && (typeof (user.gender) === 'string');
+}
+
+//перевірка на валідність users
+function checkValid(users) {
+    return users.filter((user) => isValid(user));
+}
+
+
+
+//перевірка на валідність String
+function validString(str) {
+    if (!str || typeof (str) != 'string') {
+        return false;
+    }
+    if (str.charAt(0) === str.charAt(0).toLowerCase()) {
+        return false;
+    }
+    return true;
+}
+//перевірка на валідність age
+function checkAge(age) {
+    if (typeof (age) != 'number' || age < 0) {
```

Comment:
I think it's usually more clear to check for what you want to get, instead of the opposite. So:
```suggestion
    if (typeof age === 'number' && age >= 0) { return true } else { return false }
```

Same in other places.